### PR TITLE
Feature/aeid 84b

### PR DESCRIPTION
**Pull request by malakai97**
Generally, applies my suggestions from PR #2.

* Condenses default branch support down to a single source of truth, the instance(.yml). 
* Adds a cli command to display or set the default branch
* Separates cli deploy revision support into the --revision option, for clarity. 

I went with this approach because the instance is not available when the command help is displayed, so we cannot display a dynamically discovered default branch for the specific instance.  This was necessary because we want the deploy command to be as simple as possible in most cases.

### CODE REVIEW

**Comment by malakai97** [comment]
This is a sensible suggestion, and I'm open to it, but let me say this first. 

My aim here was to make it obvious to naive users who want to specify a branch or revision how to do that, without requiring them to know the relationship between a commit and a branch in git.  Following the git utility's argument pattern is not particularly compelling, as it is famously inconsistent and arcane.  

tl;dr: I agree with your suggestion in principle, but in practice I think people less comfortable with git will be confused.

If you still think I should make that change, I'll make it.  

Thanks!

File: lib/fauxpaas/capistrano_deployer.rb [hunk 1]
```diff
@@ -8,9 +8,9 @@ def initialize(capfile_path, kernel = Kernel)
       @kernel = kernel
     end
 
-    def deploy(instance,branch: 'master')
+    def deploy(instance, branch: nil, revision: nil)
       instance_capfile_path = capfile_path + "#{instance.deployer_env}.capfile"
-      kernel.system("cap -f #{instance_capfile_path} #{instance.name} deploy BRANCH=#{branch}")
+      kernel.system("cap -f #{instance_capfile_path} #{instance.name} deploy BRANCH=#{revision || branch || instance.default_branch}")
```
Comment (botimer) [inline lib/fauxpaas/capistrano_deployer.rb]: Eeek! I think we have to put this branching/resolution into a method and just call it here.

File: lib/fauxpaas/cli/main.rb [hunk 2]
```diff
@@ -7,15 +7,33 @@ module CLI
...
+        Fauxpaas.deployer.deploy(instance, branch: options[:branch], revision: options[:revision])
+      end
+
+      desc "default_branch <instance> [<new_branch>]",
+        "Display or set the default branch for the instance"
+      def default_branch(instance_name, new_branch = nil)
+        instance = Fauxpaas.instance_repo.find(instance_name)
+        if new_branch
+          instance.default_branch = new_branch
+          Fauxpaas.instance_repo.save(instance)
```
Comment (botimer) [inline lib/fauxpaas/cli/main.rb]: The output is unclear from the callsite. I'd probably do a bare if and return after.

Comment (botimer) [reply lib/fauxpaas/cli/main.rb]: I'm willfully ignoring what `.save` returns. If if-else branches should have different output messages, they should be prepared here. If it should always be `instance.default_branch`, it should happen after the no-else if.

Comment (malakai97) [reply lib/fauxpaas/cli/main.rb]: I don't get your meaning.

Comment (botimer) [reply lib/fauxpaas/cli/main.rb]:
```
if new_branch
  instance.default_branch = new_branch
  Fauxpaas.instance_repo.save(instance)
end

instance.default_branch
```
or

```
if new_branch
  instance.default_branch = new_branch
  Fauxpass.instance_repo.save(instance)
  "Some successful update message"
else
  "Message containing unchanged value"
end
```
